application-confluence-migrator-pro-api/src/main/java/com/xwiki/confluencepro/internal/DefaultConfluenceMigratorManager.java

```diff
@@ -0,0 +1,204 @@
+                .bindValue("pages", pageToLog.keySet()).execute();
+
+            Map<String, List<String>> skipped = new TreeMap<>();
+            Map<String, List<String>> problematic = new TreeMap<>();
+
+            for (Object[] documentId : documentIds) {
+                String serializedDocRef = (String) documentId[0];
+                String title = (String) documentId[1];
+
+                for (LogEvent logEvent : pageToLog.get(title)) {
+                    if (logEvent.getLevel().equals(LogLevel.ERROR)) {
+                        skipped.computeIfAbsent(serializedDocRef, k -> new ArrayList<>())
+                            .add(logEvent.getFormattedMessage());
+                    } else {
+                        problematic.computeIfAbsent(serializedDocRef, k -> new ArrayList<>())
+                            .add(logEvent.getFormattedMessage());
+                    }
+                }
+            }
+            object.set("skipped", gson.toJson(skipped), context);
+            object.set("problems", gson.toJson(problematic), context);
+            object.set("otherIssues", gson.toJson(otherIssues), context);
+        } catch (QueryException e) {
+            throw new RuntimeException(e);
+        }
+    }
+
+    private Optional<PageIdentifier> getPageIdentifierArgument(LogEvent logEvent, Map<String, List<String>> otherIssues)
+    {
+        if (!logEvent.getLevel().equals(LogLevel.ERROR) && !logEvent.getLevel().equals(LogLevel.WARN)) {
+            return Optional.empty();
+        }
+        if (logEvent.getArgumentArray() == null) {
+            otherIssues.computeIfAbsent(logEvent.getLevel().toString(), k -> new ArrayList<>())
+                .add(logEvent.getFormattedMessage());
+            return Optional.empty();
+        }
+        Optional<PageIdentifier> pageIdentifier =
+            Arrays.stream(logEvent.getArgumentArray()).filter(a -> a instanceof PageIdentifier)
+                .map(a -> (PageIdentifier) a).findFirst();
+        // Do this here so we don't have to look through the arguments for each log again.
+        if (!pageIdentifier.isPresent() || pageIdentifier.get().getPageTitle().isEmpty()) {
+            otherIssues.computeIfAbsent(logEvent.getLevel().toString(), k -> new ArrayList<>())
+                .add(logEvent.getFormattedMessage());
+            return Optional.empty();
+        }
+        return pageIdentifier;
+    }
+
+    private void addToJsonList(LogEvent logEvent, List<Map<String, Object>> logList)
+    {
+        Map<String, Object> logMap = new HashMap<>();
+        logMap.put("level", logEvent.getLevel().toString());
+        logMap.put("timeStamp", logEvent.getTimeStamp());
+        logMap.put("message", logEvent.getFormattedMessage());
+        if (logEvent.getThrowable() != null) {
+            logMap.put("throwable", ExceptionUtils.getStackFrames(logEvent.getThrowable()));
+        }
+        logList.add(logMap);
```

Comment:
You are duplicating the information you have in the job status (serialized in the file system).